graviti/dataset/dataset.py

```diff
@@ -5,20 +5,37 @@
 
 """The implementation of the Graviti Dataset."""
 
-from typing import Optional
+from functools import wraps
+from typing import AbstractSet, Any, Callable, Dict, Iterator, Mapping, Optional, TypeVar
 
 from tensorbay.client.status import Status
-from tensorbay.utility.user import UserMutableMapping
 
+# from graviti.client.catalog import get_catalog
+from graviti.client.data import list_data_details
 from graviti.client.dataset import get_dataset
+from graviti.client.segment import list_segments
 from graviti.dataframe.frame import DataFrame
 from graviti.manager.branch import BranchManager
 from graviti.manager.commit import CommitManager
 from graviti.manager.draft import DraftManager
 from graviti.manager.tag import TagManager
+from graviti.utility.lazy import LazyFactory
 
+_Callable = TypeVar("_Callable", bound=Callable[..., Any])
 
-class Dataset(UserMutableMapping[str, DataFrame]):
+
+def _getter(func: _Callable) -> _Callable:
```

Comment:
Compare to decorator, will a method to check and return the `_data` be better?
ref: https://github.com/linjiX/tensorbay-python-sdk/blob/e03ae156c9db410528146b26054796171d54d1f8/tensorbay/client/lazy.py#L345